nexus/tests/integration_tests/saml.rs

```diff
@@ -35,6 +35,8 @@ pub const SAML_IDP_DESCRIPTOR_NO_KEYS: &str =
     include_str!("data/saml_idp_descriptor_no_keys.xml");
 
 // Create a SAML IdP
+//TODO(ry) this is broken on my system
```

Comment:
All these saml tests are failing on my workstation with.

```
--- STDERR:              omicron-nexus::test_all integration_tests::saml::test_post_saml_response_with_relay_state ---
log file: /tmp/test_all-cff3f951c9d1b075-test_post_saml_response_with_relay_state.10581.0.log
note: configured to log to "/tmp/test_all-cff3f951c9d1b075-test_post_saml_response_with_relay_state.10581.0.log"
DB URL: postgresql://root@[::1]:41288/omicron?sslmode=disable
DB address: [::1]:41288
log file: /tmp/test_all-cff3f951c9d1b075-test_post_saml_response_with_relay_state.10581.2.log
note: configured to log to "/tmp/test_all-cff3f951c9d1b075-test_post_saml_response_with_relay_state.10581.2.log"
func=xmlSecOpenSSLAppCertLoadBIO:file=app.c:line=1294:obj=unknown:subj=d2i_X509_bio:error=4:crypto library function failed:openssl error: error:00000000:lib(0)::reason(0)
func=xmlSecOpenSSLAppKeyFromCertLoadBIO:file=app.c:line=1014:obj=unknown:subj=xmlSecOpenSSLAppCertLoadBIO:error=1:xmlsec library function failed:
func=xmlSecOpenSSLAppKeyLoadBIO:file=app.c:line=414:obj=unknown:subj=xmlSecOpenSSLAppKeyFromCertLoadBIO:error=1:xmlsec library function failed:
func=xmlSecOpenSSLAppKeyLoadMemory:file=app.c:line=306:obj=unknown:subj=xmlSecOpenSSLAppKeyLoadBIO:error=1:xmlsec library function failed:
thread 'integration_tests::saml::test_post_saml_response_with_relay_state' panicked at nexus/tests/integration_tests/saml.rs:1339:6:
expected success: expected status code 303 See Other, found 400 Bad Request
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
WARN: dropped CockroachInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)
WARN: temporary directory leaked: "/tmp/.tmp5scdly"
        If you would like to access the database for debugging, run the following:

        # Run the database
        cargo run --bin omicron-dev db-run --no-populate --store-dir "/tmp/.tmp5scdly/data"
        # Access the database. Note the port may change if you run multiple databases.
        cockroach sql --host=localhost:32221 --insecure
WARN: dropped ClickHouseInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)

   Canceling due to test failure
------------
     Summary [  18.944s] 1 test run: 0 passed, 1 failed, 1072 skipped
     SIGSEGV [   5.898s] omicron-nexus::test_all integration_tests::saml::test_post_saml_response_with_relay_state
```

I'm running stock helios that I update regularly.

Reply:
Here is what I'm running locally

```
ry@masaka:~$ uname -a
SunOS masaka 5.11 helios-2.0.22500 i86pc i386 i86pc
```
```
ry@masaka:~$ cat /etc/versions/build
heads/stlouis-0-g9318fd43fb
```

Reply:
This appears to have been due use of the `mold` linker on illumos.